test/orange/CMakeLists.txt

```diff
@@ -30,6 +30,14 @@ if(CELERITAS_USE_Geant4)
   celeritas_get_g4libs(_g4_geo_libs geometry)
 endif()
 
+if(CELERITAS_USE_Geant4 AND CELERITAS_REAL_TYPE STREQUAL "double")
+  set(_use_g4org TRUE)
+  set(_needs_g4org)
+else()
+  set(_use_g4org FALSE)
```

Comment:
Yes, it can handle them on its own but  right now there are some type issues for the double/float. I agree about using doubles for the CPU setup code and floats for runtime code. However, that would require some extra templating in the code (for classes that are accessed both at setup time and runtime) and so I'm not sure if that's worth the near-term effort.